CMake/RealmCore.cmake

```diff
@@ -40,7 +40,7 @@ endif()
 
 function(use_realm_core version_or_path_to_source)
     if("${version_or_path_to_source}" MATCHES "^[0-9]+(\\.[0-9])+")
-        if(APPLE OR REALM_PLATFORM STREQUAL "Android")
+        if(APPLE OR LINUX OR REALM_PLATFORM STREQUAL "Android")
```

Comment:
I think it is, but the way we define `LINUX` is kind of hacky and overly broad. I think it should be `CMAKE_SYSTEM_NAME STREQUAL "Linux"` but I'm not sure that would actually include Android if you were cross-compiling on a Darwin host. Need to check the toolchain file.